---
title: "Assign clone barcodes to cells in scRNAseq data using CloneDetective"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Assign clone barcodes to cells in scRNAseq data using CloneDetective}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Introduction

This vignette focuses on using CloneDetective to analyse scRNAseq data.
Specifically, we will look at how many clone barcodes are detected per cell,
and look at how to assign clone barcodes to cells.

# Setup

In addition to CloneDetective weâ€™ll load the following packages:

* **data.table**, to read in CSV file.
* **scater**, to simulate SingleCellExperiment object.

```{r setup}
library(CloneDetective)
library(scater)
library(data.table)
```

# Simulate scRNAseq data

For this vignette, we will use a simulated cell by gene matrix generated
using the `scater` package.

```{r}
set.seed(42)

sce <- mockSCE(ncells = 2000, ngenes = 1000)
# use the colnames as the 10x cell barcode
colData(sce)$Barcode <- colnames(sce)

sce
```

## Loading actual cell by gene matrix.

For your own dataset, you can load them up using the [DropletUtils](https://bioconductor.org/packages/release/bioc/html/DropletUtils.html)
package.

Assuming you have the `barcode.tsv.gz`, `features.tsv.gz`, `matrix.mtx.gz` files
in a folder `filtered_feature_bc_matrix`.
You can load them up into a SingleCellExperiment object like so.

```{r eval=FALSE}
sce <- read10xCounts("filtered_feature_bc_matrix")
```

At the end, you should have in colData of the object,
a column called Barcode which represents the 10x cell barcode.

# Load the clone barcode data

Assuming you have used [NextClone](https://github.com/phipsonlab/NextClone) to extract 
the clone barcodes from your BAM file, 
you will now have a CSV file where each row corresponds to a read that belongs to
a cell and mapped to a clone barcode.

If you have used other pipeline to extract (which you are more than welcomed to),
you will have to wrangle the output such that you get the aforementioned CSV file.

For the purpose of this vignette, I have randomly sampled 2000 cells from the
scRNAseq data we used in our paper.

Let's load it up using `data.table` and take a peek inside using the `head` function.
The data is included in the package in `extdata` folder which we can located using
`system.file` function.

```{r}
cell_clone_bcode_dt <- fread(
  system.file("extdata", "sc_clone_barcodes_subsampled.csv", 
              package = "CloneDetective")
)


head(cell_clone_bcode_dt)
```

Remember, each row is a read.
Let's quickly go through what the columns are:

* **CellBarcode**: 10x barcode which uniquely identify your cells. Since this is a 
simulated data, the Barcode is just "Cell_xxx". For real data, your CellBarcode
should look like "ATTTGG-1" (something along those lines).
* **CloneBarcode**: the clone barcode detected.
* **SourceBAMFile**: the BAM file the read comes from.
* **ReadId**: the read ID.
* **UMI**: UMI barcode for the read.
* **FlankEditDist**: the edit distance for the 5' and 3' adapter for the read.
This is generated by Flexiplex.
* **BarcodeEditDist**: the edit distance when mapping the read's sequence
to the clone barcode.
This is generated by Flexiplex.

# Generate cell by clone matrix.

We can use CloneDetective to generate a table that looks a lot like
a cell by clone matrix where each row represents a cell, a clone barcode,
and the number of reads detected for that cell and clone barcode.

```{r}
cell_by_clone_mat = generate_cell_clone_barcode_matrix(
  cell_clone_bcode_dt = cell_clone_bcode_dt,
  cell_bcode_col = "CellBarcode",
  clone_bcode_col = "CloneBarcode",
  umi_col = "UMI",
  umi_clone_consensus_threshold = 0.7
)
```

The parameters ending with `_col` denotes the column that represents the cell barcode
(`cell_bcode_col`), the clone barcode (`clone_bcode_col`), and the UMI (`umi_col`).

The `umi_clone_consensus_threshold` represents the proportion threshold of reads for collapsing UMIs when computing the cell-by-clone matrix.
The cell-by-clone matrix construction first collapses reads with the same cell and UMI barcodes 
For a group of reads have the same cell barcode and UMI barcode, if the reads are mapped to several clone barcodes, 
by default, they are collapsed into one read and assigned to the clone barcode comprising 70%
or more of its group's reads.
This threshold modifiable using this `umi_clone_consensus_threshold` parameter. 
To apply the default threshold of 70%, set this parameter to 0.7.

This is what the cell by clone matrix looks like.

```{r}
head(cell_by_clone_mat)
```

For each cell and clone barcode, you will see how many reads were detected.

# Drawing Treemap

A Treemap is useful for visualising the distribution of cells (as absolute number and proportion) 
according to the number of detected clone barcodes within each cell.
It cells into groups based on the number of clone barcodes detected per cell, 
ranging from cells with no detected barcodes to cells with ten or more barcodes detected. 

To draw the treemap, we have to give it the cell barcode to draw the plot for.
For this, we will use all the cell barcode in our SingleCellExperiment object.

```{r}
valid_cells_10x <- colData(sce)$Barcode
plt <- draw_treemap(
    cell_by_clone_matrix = cell_by_clone_mat,
    valid_cells_bcodes = valid_cells_10x
)
plt
```


Ideally, each cell in a scRNAseq experiment should only have a single clone barcode detected. 
However, it's common to find cells with either no or multiple clone barcodes detected. 

# Assigning clones to cells

To do this, we can use the `assign_and_embed_clones` function like so.

```{r}
sce_with_clone <- assign_and_embed_clones(
    cell_by_gene_mat = sce,
    cell_clone_reads_dt = cell_clone_bcode_dt,
)
colData(sce_with_clone)
```

You will notice two new columns in your SingleCellExperiment object.
`clone_barcode` denotes the clone barcode detected for the cell.
NA means no clone barcode was detected.
`clone_barcode_criteria` refers to how the clone barcode assignment was done.

This is the criteria we used to do the clone barcode assignment (in order):

1. Cells with exactly one clone barcode are assigned to their respective clones. 
2. For cells with multiple clone barcodes detected, we resolved the assignment by evaluating the read abundance. 
Cells were first assigned to the clone barcode that accounted for over half of the detected PCR deduplicated reads. 
3. The remaining cells which did not meet the above criterion underwent further analysis based on average barcode edit distances. 
Here, cells were assigned to the clone barcode with the smallest average edit distance. 
In cases where distances were identical, the tie was broken in favour of the clone barcode with the higher read proportion.



