<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Assign clones to cells — assign_and_embed_clones • CloneDetective</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Assign clones to cells — assign_and_embed_clones"><meta property="og:description" content="This function assigns clone barcodes to cells and optionally embeds the
assignment into a SingleCellExperiment object. It calculates a cell-by-clone
matrix from a provided data.table where each row corresponds to a read
associated with a cell, UMI, and clone barcode.
See details for the assignment method."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CloneDetective</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/index.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Other Packages
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="https://phipsonlab.github.io/NextClone/" class="external-link">NextClone</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/phipsonlab/CloneDetective" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Assign clones to cells</h1>
    
    <div class="hidden name"><code>assign_and_embed_clones.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function assigns clone barcodes to cells and optionally embeds the
assignment into a SingleCellExperiment object. It calculates a cell-by-clone
matrix from a provided data.table where each row corresponds to a read
associated with a cell, UMI, and clone barcode.
See details for the assignment method.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">assign_and_embed_clones</span><span class="op">(</span></span>
<span>  <span class="va">cell_by_gene_mat</span>,</span>
<span>  <span class="va">cell_clone_reads_dt</span>,</span>
<span>  cell_bcode_col <span class="op">=</span> <span class="st">"CellBarcode"</span>,</span>
<span>  barcode_edit_dist_col <span class="op">=</span> <span class="st">"BarcodeEditDist"</span>,</span>
<span>  clone_bcode_col <span class="op">=</span> <span class="st">"CloneBarcode"</span>,</span>
<span>  umi_col <span class="op">=</span> <span class="st">"UMI"</span>,</span>
<span>  most_dominant_threshold <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  umi_clone_consensus_threshold <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>  embed_to_mat <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>cell_by_gene_mat</dt>
<dd><p>A SingleCellExperiment object containing the
cell-by-gene matrix. It must have a 'Barcode' column in 'colData' that
uniquely identifies each cell. The barcodes should match those in the
`cell_clone_reads_dt` data table.</p></dd>


<dt>cell_clone_reads_dt</dt>
<dd><p>A data.table object representing the reads. Each
row includes information about a cell, UMI, and clone barcode. For data
produced by NextClone, use `fread` from the data.table package and pass the
resulting object to this parameter.</p></dd>


<dt>cell_bcode_col</dt>
<dd><p>Name of the column in `cell_clone_reads_dt` that
indicates the cell barcode for each read.</p></dd>


<dt>barcode_edit_dist_col</dt>
<dd><p>Name of the column in `cell_clone_reads_dt` that
indicates the edit distance for the clone barcode of each read.</p></dd>


<dt>clone_bcode_col</dt>
<dd><p>Name of the column in `cell_clone_reads_dt` that
specifies the clone barcode for each read.</p></dd>


<dt>umi_col</dt>
<dd><p>Name of the column in `cell_clone_reads_dt` that specifies the
UMI barcode for each read.</p></dd>


<dt>most_dominant_threshold</dt>
<dd><p>A numeric value between 0 and 1 specifying the
proportion threshold of reads that must be associated with a clone barcode to
assign it to a cell in cases where multiple clone barcodes are detected.</p></dd>


<dt>umi_clone_consensus_threshold</dt>
<dd><p>A numeric value between 0 and 1 specifying
the proportion threshold of reads for collapsing UMIs when computing the
cell-by-clone matrix.
See details for more information.</p></dd>


<dt>embed_to_mat</dt>
<dd><p>A boolean indicating whether to embed the clone barcode
assignment directly into the `cell_by_gene_mat` object.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Depending on the `embed_to_mat` parameter, this function returns either
an updated SingleCellExperiment object with clone barcode assignments embedded
or a data.table with the cell-clone assignments.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Clone barcode assignment to cells follows a tiered approach:
Cells with a single detected clone barcode are straightforwardly assigned to that clone.
In cases where multiple clone barcodes are present, the most dominant clone barcode
(constituting over 50% of reads, adjustable via `most_dominant_threshold`) is assigned.
Remaining cells, where no clone barcode is sufficiently dominant, are assigned based
on the lowest average barcode edit distance. If edit distances are equal, the clone
barcode with the higher read count prevails.
To use the default 50% threshold, set `most_dominant_threshold` to 0.5.</p>
<p>The cell-by-clone matrix construction first collapses reads with the same cell
and UMI barcodes
For a group of reads have the same cell barcode and UMI barcode,
if the reads are mapped to several clone barcodes,
by default, they are collapsed into one read and assigned to the clone barcode
comprising 70% or more of its group's reads.
This threshold modifiable by the `umi_clone_consensus_threshold` parameter.
To apply the default threshold of 70%, set this parameter to 0.7.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: SingleCellExperiment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: SummarizedExperiment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: MatrixGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: matrixStats</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘MatrixGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:matrixStats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colWeightedMeans, colWeightedMedians, colWeightedSds,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowWeightedSds, rowWeightedVars</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: stats4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BiocGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘BiocGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:stats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     IQR, mad, sd, var, xtabs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.data.frame, basename, cbind, colnames, dirname, do.call,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     tapply, union, unique, unsplit, which.max, which.min</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: S4Vectors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘S4Vectors’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:utils’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     findMatches</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     I, expand.grid, unname</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: IRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomeInfoDb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Biobase</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Welcome to Bioconductor</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Vignettes contain introductory material; view with</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘Biobase’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:MatrixGenerics’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowMedians</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:matrixStats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     anyMissing, rowMedians</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: scuttle</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: ggplot2</span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘data.table’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:SummarizedExperiment’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     shift</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:GenomicRanges’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     shift</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:IRanges’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     shift</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:S4Vectors’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     first, second</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/mockSCE.html" class="external-link">mockSCE</a></span><span class="op">(</span>ncells <span class="op">=</span> <span class="fl">4</span>, ngenes <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">Barcode</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">cell_clone_reads_dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    CellBarcode <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Cell_001"</span>, <span class="fl">6</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Cell_002"</span>, <span class="fl">7</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Cell_003"</span>, <span class="fl">4</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Cell_004"</span>, <span class="fl">5</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="op">)</span>,</span></span>
<span class="r-in"><span>    CloneBarcode <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"AA"</span>, <span class="fl">6</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"BB1"</span>, <span class="fl">4</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"BB2"</span>, <span class="fl">3</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"CC1"</span>, <span class="fl">2</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"CC2"</span>, <span class="fl">2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"CC1"</span>, <span class="fl">2</span><span class="op">)</span>, <span class="st">"DD1"</span>, <span class="st">"DD2"</span>, <span class="st">"DD3"</span></span></span>
<span class="r-in"><span>    <span class="op">)</span>,</span></span>
<span class="r-in"><span>    BarcodeEditDist <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0L</span>, <span class="fl">6</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0L</span>, <span class="fl">7</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">cell_clone_reads_dt</span><span class="op">[</span>, <span class="va">UMI</span> <span class="op">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">cell_clone_reads_dt</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     CellBarcode CloneBarcode BarcodeEditDist UMI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:    Cell_001           AA               0   1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:    Cell_001           AA               0   2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:    Cell_001           AA               0   3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:    Cell_001           AA               0   4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:    Cell_001           AA               0   5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:    Cell_001           AA               0   6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:    Cell_002          BB1               0   7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:    Cell_002          BB1               0   8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:    Cell_002          BB1               0   9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:    Cell_002          BB1               0  10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:    Cell_002          BB2               0  11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:    Cell_002          BB2               0  12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:    Cell_002          BB2               0  13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:    Cell_003          CC1               0  14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:    Cell_003          CC1               1  15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:    Cell_003          CC2               2  16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:    Cell_003          CC2               1  17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:    Cell_004          CC1               0  18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:    Cell_004          CC1               0  19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:    Cell_004          DD1               0  20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21:    Cell_004          DD2               0  21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22:    Cell_004          DD3               0  22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     CellBarcode CloneBarcode BarcodeEditDist UMI</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">sce_with_clone</span> <span class="op">&lt;-</span> <span class="fu">assign_and_embed_clones</span><span class="op">(</span></span></span>
<span class="r-in"><span>    cell_by_gene_mat <span class="op">=</span> <span class="va">sce</span>,</span></span>
<span class="r-in"><span>    cell_clone_reads_dt <span class="op">=</span> <span class="va">cell_clone_reads_dt</span>,</span></span>
<span class="r-in"><span>    cell_bcode_col <span class="op">=</span> <span class="st">"CellBarcode"</span>,</span></span>
<span class="r-in"><span>    barcode_edit_dist_col <span class="op">=</span> <span class="st">"BarcodeEditDist"</span>,</span></span>
<span class="r-in"><span>    clone_bcode_col <span class="op">=</span> <span class="st">"CloneBarcode"</span>,</span></span>
<span class="r-in"><span>    umi_col <span class="op">=</span> <span class="st">"UMI"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">colData</span><span class="op">(</span><span class="va">sce_with_clone</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 4 rows and 6 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          Mutation_Status  Cell_Cycle   Treatment     Barcode clone_barcode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;   &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cell_001        positive          G1      treat1    Cell_001            AA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cell_002        positive          G0      treat2    Cell_002           BB1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cell_003        negative          G0      treat1    Cell_003           CC1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cell_004        positive         G2M      treat2    Cell_004           CC1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               clone_barcode_criteria</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             &lt;factor&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cell_001 single_clone               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cell_002 dominant_clone_moreThan_0_5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cell_003 clone_from_edit_distance   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cell_004 clone_from_edit_distance   </span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Givanna Putri.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

