<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analyse DNA-seq data using CloneDetective • CloneDetective</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analyse DNA-seq data using CloneDetective">
<meta property="og:description" content="CloneDetective">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CloneDetective</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/index.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Other Packages
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="https://github.com/phipsonlab/NextClone" class="external-link">NextClone</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/phipsonlab/CloneDetective" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analyse DNA-seq data using CloneDetective</h1>
            
      
      
      <div class="hidden name"><code>analyse_dnaseq_data.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette focuses on using the package to analyse DNA-seq data.
DNA-seq data refers to lineage tracing DNA barcoding data which
exclusively sequences clone barcode reads using Next Generation
Sequencing.</p>
<p>We will go through how to analyse clone barcode distributions from
DNA-seq data which can inform the design of future scRNAseq
experiments.</p>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<p>In addition to CloneDetective we’ll load the following packages:</p>
<ul>
<li>
<strong>data.table</strong>, to read in CSV file and manipulate
them.</li>
<li>
<strong>ggplot2</strong>, in case we want to customise the ggplot
objects created by CloneDetective.</li>
<li>
<strong>scales</strong>, in case we want to customise the scale of
the ggplot objects created by CloneDetective.</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://phipsonlab.github.io/CloneDetective/">CloneDetective</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://scales.r-lib.org" class="external-link">scales</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-the-dna-seq-data">Load the DNA-seq data<a class="anchor" aria-label="anchor" href="#load-the-dna-seq-data"></a>
</h2>
<p>Before we delve into investigating what our DNA-seq data looks like,
we have to load it in first.</p>
<p>Assuming you have used <a href="https://github.com/phipsonlab/NextClone" class="external-link">NextClone</a> to extract
the clone barcodes from your raw FASTQ files, you will now have a CSV
file where each row corresponds to a unique clone barcode and its
frequency in the sample.</p>
<p>If you have used other pipeline to extract (which you are more than
welcomed to), you will have to wrangle the output such that you get the
aforementioned CSV file.</p>
<p>For the purpose of this vignette, I have randomly sampled 4000 clone
barcodes from the 8k and 10k DNA-seq data we used in our paper.</p>
<p>Let’s load it up using <code>data.table</code> and take a peek inside
using the <code>head</code> function. The data is included in the
package in <code>extdata</code> folder which we can located using
<code>system.file</code> function.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clones_nxclone</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"dnaseq_clone_barcode_counts_subsampled.csv"</span>, </span>
<span>              package <span class="op">=</span> <span class="st">"CloneDetective"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">clones_nxclone</span><span class="op">)</span></span>
<span><span class="co">#&gt;    sample_name        known_barcode read_count</span></span>
<span><span class="co">#&gt; 1:         10k TCGGACGCTGGTGAATCTTT        190</span></span>
<span><span class="co">#&gt; 2:         10k GAAGTATCTGGGTTTGGAAG        291</span></span>
<span><span class="co">#&gt; 3:         10k TCTTGAGGGGAGCCAGACCG         91</span></span>
<span><span class="co">#&gt; 4:         10k TTTCAGGGCGTGGATACCAG        404</span></span>
<span><span class="co">#&gt; 5:         10k GAAGGGGAGGAGTCATGTCG        797</span></span>
<span><span class="co">#&gt; 6:         10k TCTAGCAGAGGAAGAGAAGG         15</span></span></code></pre></div>
<p>Let’s quickly go through what the columns are:</p>
<ul>
<li>
<strong>sample_name</strong>: the sample the clone barcode came
from. We have 2 samples for this dataset, 8k and 10k.</li>
<li>
<strong>known_barcode</strong>: the clone barcodes in our data.</li>
<li>
<strong>read_count</strong>: the number of reads detected for each
clone. Since we only sequence the clone barcode, you can treat this as
the number of cells associated with a given clone barcode.</li>
</ul>
</div>
<div class="section level2">
<h2 id="analyse-clone-barcode-distributions">Analyse clone barcode distributions<a class="anchor" aria-label="anchor" href="#analyse-clone-barcode-distributions"></a>
</h2>
<p>CloneDetective offers functions to count the number of clone barcodes
in the dataset and to filter out clones with low abundance based on a
predefined cell count threshold.</p>
<p>To do this, we have to first specify the filtering thresholds,
i.e. the numeric vector listing the minimum cell counts required for a
clone to be included in the count. For this example, we will set the
thresholds to 1, 20, 200, 1000 to see how many clones we get with at
least 1, 20, 200, 1000 cells.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">thresholds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">20</span>, <span class="fl">200</span>, <span class="fl">1000</span><span class="op">)</span></span></code></pre></div>
<p>Then we will count how many clones with 1, 20, 200, 1000 cells.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_barcodes_nxclone</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/count_retained_clones.html">count_retained_clones</a></span><span class="op">(</span></span>
<span>    count_data <span class="op">=</span> <span class="va">clones_nxclone</span>,</span>
<span>    thresholds <span class="op">=</span> <span class="va">thresholds</span>,</span>
<span>    grouping_col <span class="op">=</span> <span class="st">"sample_name"</span>,</span>
<span>    count_column <span class="op">=</span> <span class="st">"read_count"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The <code>grouping_col</code> parameter essentially tells the
function to repeat the counting procedure for each unique value in the
<strong>sample_name</strong> column of our clone data
(<code>clones_nxclone</code>). The <code>count_column</code> tells the
function which column contains the number of cells detected for each
clone.</p>
<p>Let’s take a peak at the result.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_barcodes_nxclone</span></span>
<span><span class="co">#&gt;    sample_name at_least_1_cells at_least_20_cells at_least_200_cells</span></span>
<span><span class="co">#&gt; 1:         10k             4000              3510               2059</span></span>
<span><span class="co">#&gt; 2:          8k             4000              3578               2257</span></span>
<span><span class="co">#&gt;    at_least_1000_cells</span></span>
<span><span class="co">#&gt; 1:                 247</span></span>
<span><span class="co">#&gt; 2:                 355</span></span></code></pre></div>
<p>Each row is a sample, and each column tells you how many clones we
get with at least 1, 20, 200, 1000 cells. Remember, we are using
sub-sampled DNA-seq data, that is why the <code>at_least_1_cells</code>
column for both samples have exactly 4,000 clones.</p>
<p>This information is useful to see how many reasonably sized clones we
get in our DNA-seq data. Often, it is not even worthwhile keeping clones
that have too little cells. To remove them, we can use the
<code>remove_clones_below_threshold</code> function.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clones_nxclone_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_clones_below_threshold.html">remove_clones_below_threshold</a></span><span class="op">(</span></span>
<span>    count_data <span class="op">=</span> <span class="va">clones_nxclone</span>,</span>
<span>    threshold <span class="op">=</span> <span class="fl">20</span>,</span>
<span>    count_column <span class="op">=</span> <span class="st">"read_count"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Let’s see how many clones were removed using the <code>table</code>
function which is a neat little function that count the number of
occurences in a vector.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">clones_nxclone_filtered</span><span class="op">$</span><span class="va">sample_name</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  10k   8k </span></span>
<span><span class="co">#&gt; 3510 3578</span></span></code></pre></div>
<p>While these are all super useful, it is even better if we can look at
the proportion of cells in each clone (against the total number of cells
in the sample), and illustrate this as an ordered abundance plot. Such
plots are insightful for determining whether clones proliferated
uniformly or if some clones were more dominant, potentially outcompeting
and restricting the expansion of others.</p>
<p>We can draw the plot using <code>draw_ordered_abundance_plot</code>,
but before we do that, we need to convert our the cell count to
proportion. We can do this using
<code>convert_count_to_proportion</code> function. Let’s run all of that
and add a bit of customisation to our plot to make it look pretty.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clones_nxclone_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/convert_count_to_proportion.html">convert_count_to_proportion</a></span><span class="op">(</span></span>
<span>  count_data <span class="op">=</span> <span class="va">clones_nxclone_filtered</span>,</span>
<span>  grouping_col <span class="op">=</span> <span class="st">"sample_name"</span>,</span>
<span>  count_column <span class="op">=</span> <span class="st">"read_count"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/draw_ordered_abundance_plot.html">draw_ordered_abundance_plot</a></span><span class="op">(</span></span>
<span>  count_data <span class="op">=</span> <span class="va">clones_nxclone_filtered</span>, </span>
<span>  facet_column <span class="op">=</span> <span class="st">"sample_name"</span>,</span>
<span>  y_axis_column <span class="op">=</span> <span class="st">"read_proportion"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">plt</span> <span class="op">&lt;-</span> <span class="va">plt</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">0.5</span>, colour<span class="op">=</span><span class="st">'red'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    title <span class="op">=</span> <span class="st">"Proportion of cells assigned to each barcode"</span>,</span>
<span>    subtitle <span class="op">=</span> <span class="st">"Barcode IDs are numerically assigned in order of read proportion"</span>,</span>
<span>    x <span class="op">=</span> <span class="st">"Numerical barcode ID"</span>,</span>
<span>    y <span class="op">=</span> <span class="st">"Proportion of cells per sample"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">plt</span></span></code></pre></div>
<p><img src="analyse_dnaseq_data_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>From this plot, you can see that the 10k sample looks even, which
means the clones are expanding evenly. However, for the 8k sample, we
have one clone (one outlier dot at around 0.015 mark at the y-axis)
which seems to have expanded much more than others. The proportion of
cells belonging to this clone is much much more than others.</p>
<p>For the future, it may be best to not use the 8k sample because of
this dominant clone restricting the space available for others to
grow.</p>
</div>
<div class="section level2">
<h2 id="using-dna-seq-data-to-plan-future-scrnaseq-experiment">Using DNA-seq data to plan future scRNAseq experiment<a class="anchor" aria-label="anchor" href="#using-dna-seq-data-to-plan-future-scrnaseq-experiment"></a>
</h2>
<p>We can use the calculated cell proportions above to forecast clonal
abundance for future scRNAseq experiments. This is helpful for
estimating the number of cells to sequence for a given clonal library in
order to maximise the chance of obtaining a reasonable representation of
clones. For example, say we plan to sequence 10,000 or 20,000 cells. Can
we reasonably estimate how many cells will be obtained per clone? Yes we
can.</p>
<p>First thing first though, we need to do the estimation. We can do
this using the <code>project_clones</code> function.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clones_nxclone_proportion</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/project_clones.html">project_clones</a></span><span class="op">(</span></span>
<span>  count_data <span class="op">=</span> <span class="va">clones_nxclone</span>,</span>
<span>  grouping_col <span class="op">=</span> <span class="st">"sample_name"</span>,</span>
<span>  count_column <span class="op">=</span> <span class="st">"read_count"</span>,</span>
<span>  project_amnt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10000</span>, <span class="fl">20000</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>You may have noticed by now that many functions in the package use
the same parameter names. They indeed do, and we intentionally do this
so to keep things simple. The only new parameter here is
<code>project_amnt</code> which if you look at the values, you may have
guessed what they represent, the number of cells to project our cell
proportions to.</p>
<p>Let’s take quick peek into this.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">clones_nxclone_proportion</span><span class="op">)</span></span>
<span><span class="co">#&gt;    sample_name        known_barcode read_count read_count_proportion</span></span>
<span><span class="co">#&gt; 1:         10k TCGGACGCTGGTGAATCTTT        190          1.425434e-04</span></span>
<span><span class="co">#&gt; 2:         10k GAAGTATCTGGGTTTGGAAG        291          2.183165e-04</span></span>
<span><span class="co">#&gt; 3:         10k TCTTGAGGGGAGCCAGACCG         91          6.827081e-05</span></span>
<span><span class="co">#&gt; 4:         10k TTTCAGGGCGTGGATACCAG        404          3.030924e-04</span></span>
<span><span class="co">#&gt; 5:         10k GAAGGGGAGGAGTCATGTCG        797          5.979322e-04</span></span>
<span><span class="co">#&gt; 6:         10k TCTAGCAGAGGAAGAGAAGG         15          1.125343e-05</span></span>
<span><span class="co">#&gt;    projected_10000_confidence_1 projected_20000_confidence_1</span></span>
<span><span class="co">#&gt; 1:                    1.4254344                    2.8508688</span></span>
<span><span class="co">#&gt; 2:                    2.1831653                    4.3663306</span></span>
<span><span class="co">#&gt; 3:                    0.6827081                    1.3654161</span></span>
<span><span class="co">#&gt; 4:                    3.0309237                    6.0618473</span></span>
<span><span class="co">#&gt; 5:                    5.9793222                   11.9586444</span></span>
<span><span class="co">#&gt; 6:                    0.1125343                    0.2250686</span></span></code></pre></div>
<p>We have 2 extra columns now (the
<strong>read_count_proportion</strong> column was computed using
<code>convert_count_to_proportion</code> in the previous section). Each
column represents the estimated number of cells if you sequence 10,000
or 20,000 cells.</p>
<p>Note, for this estimation, we assume all the cells will get a clone
barcode. Chances are this won’t happen, and you will probably get 80% to
90% of cells getting clone barcode. We can add this information in as
the <code>confidence_threshold</code> parameter, and redo the
estimation. Note that <code>confidence_threshold</code> takes value
between 0 to 1, so you will need to convert the percentage (if 80% then
it is 80/100 = 0.8).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clones_nxclone_proportion</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/project_clones.html">project_clones</a></span><span class="op">(</span></span>
<span>  count_data <span class="op">=</span> <span class="va">clones_nxclone</span>,</span>
<span>  grouping_col <span class="op">=</span> <span class="st">"sample_name"</span>,</span>
<span>  count_column <span class="op">=</span> <span class="st">"read_count"</span>,</span>
<span>  project_amnt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10000</span>, <span class="fl">20000</span><span class="op">)</span>,</span>
<span>  confidence_threshold <span class="op">=</span> <span class="fl">0.8</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">clones_nxclone_proportion</span><span class="op">)</span></span>
<span><span class="co">#&gt;    sample_name        known_barcode read_count read_count_proportion</span></span>
<span><span class="co">#&gt; 1:         10k TCGGACGCTGGTGAATCTTT        190          1.425434e-04</span></span>
<span><span class="co">#&gt; 2:         10k GAAGTATCTGGGTTTGGAAG        291          2.183165e-04</span></span>
<span><span class="co">#&gt; 3:         10k TCTTGAGGGGAGCCAGACCG         91          6.827081e-05</span></span>
<span><span class="co">#&gt; 4:         10k TTTCAGGGCGTGGATACCAG        404          3.030924e-04</span></span>
<span><span class="co">#&gt; 5:         10k GAAGGGGAGGAGTCATGTCG        797          5.979322e-04</span></span>
<span><span class="co">#&gt; 6:         10k TCTAGCAGAGGAAGAGAAGG         15          1.125343e-05</span></span>
<span><span class="co">#&gt;    projected_10000_confidence_0_8 projected_20000_confidence_0_8</span></span>
<span><span class="co">#&gt; 1:                              1                              1</span></span>
<span><span class="co">#&gt; 2:                              2                              3</span></span>
<span><span class="co">#&gt; 3:                              1                              1</span></span>
<span><span class="co">#&gt; 4:                              3                              6</span></span>
<span><span class="co">#&gt; 5:                              4                             10</span></span>
<span><span class="co">#&gt; 6:                              0                              0</span></span></code></pre></div>
<p>We can no visualise this as ordered abundance plot to see what the
clone abundance look like.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/draw_ordered_abundance_plot.html">draw_ordered_abundance_plot</a></span><span class="op">(</span></span>
<span>  count_data <span class="op">=</span> <span class="va">clones_nxclone_proportion</span>,</span>
<span>  y_axis_column <span class="op">=</span> <span class="st">'projected_20000_confidence_0_8'</span>,</span>
<span>  facet_column <span class="op">=</span> <span class="st">'sample_name'</span></span>
<span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.5</span>, colour<span class="op">=</span><span class="st">'forestgreen'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    y <span class="op">=</span> <span class="st">'Number of cells'</span>,</span>
<span>    title <span class="op">=</span> <span class="st">'Number of cells assigned to each barcode'</span>,</span>
<span>    subtitle <span class="op">=</span> <span class="st">'Cell counts computed after projection to 20,000 cells'</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">plt</span></span></code></pre></div>
<p><img src="analyse_dnaseq_data_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>And for sequencing 10,000 cells?</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/draw_ordered_abundance_plot.html">draw_ordered_abundance_plot</a></span><span class="op">(</span></span>
<span>  count_data <span class="op">=</span> <span class="va">clones_nxclone_proportion</span>,</span>
<span>  y_axis_column <span class="op">=</span> <span class="st">'projected_10000_confidence_0_8'</span>,</span>
<span>  facet_column <span class="op">=</span> <span class="st">'sample_name'</span></span>
<span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.5</span>, colour<span class="op">=</span><span class="st">'dodgerblue'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    y <span class="op">=</span> <span class="st">'Number of cells'</span>,</span>
<span>    title <span class="op">=</span> <span class="st">'Number of cells assigned to each barcode'</span>,</span>
<span>    subtitle <span class="op">=</span> <span class="st">'Cell counts computed after projection to 10,000 cells'</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">plt</span></span></code></pre></div>
<p><img src="analyse_dnaseq_data_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p>We can also estimate the number of clones that will contain at least
a specified minimum number of cells using
<code>count_retained_clones</code> function used in the previous
section.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">thresholds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">30</span>, <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="va">proj_n_clones_retained</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/count_retained_clones.html">count_retained_clones</a></span><span class="op">(</span></span>
<span>  count_data <span class="op">=</span> <span class="va">clones_nxclone_proportion</span>,</span>
<span>  thresholds <span class="op">=</span> <span class="va">thresholds</span>,</span>
<span>  grouping_col <span class="op">=</span> <span class="st">"sample_name"</span>,</span>
<span>  count_column <span class="op">=</span> <span class="st">"projected_10000_confidence_0_8"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">proj_n_clones_retained</span></span>
<span><span class="co">#&gt;    sample_name at_least_10_cells at_least_15_cells at_least_20_cells</span></span>
<span><span class="co">#&gt; 1:         10k                67                14                 2</span></span>
<span><span class="co">#&gt; 2:          8k                62                12                 2</span></span>
<span><span class="co">#&gt;    at_least_25_cells at_least_30_cells</span></span>
<span><span class="co">#&gt; 1:                 0                 0</span></span>
<span><span class="co">#&gt; 2:                 1                 1</span></span></code></pre></div>
<p>As well as the cumulative number of cells for the top say 200 most
abundant clones. This is particularly useful if we want to do
differential abundance test for the clones.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_threshold</span> <span class="op">&lt;-</span> <span class="fl">200</span></span>
<span><span class="va">top_barcodes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_top_barcodes_and_cum_sum.html">get_top_barcodes_and_cum_sum</a></span><span class="op">(</span></span>
<span>  count_data <span class="op">=</span> <span class="va">clones_nxclone_proportion</span>,</span>
<span>  count_column <span class="op">=</span> <span class="st">"projected_10000_confidence_0_8"</span>,</span>
<span>  grouping_col <span class="op">=</span> <span class="st">"sample_name"</span>,</span>
<span>  top_threshold <span class="op">=</span> <span class="va">top_threshold</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">top_barcodes</span><span class="op">)</span></span>
<span><span class="co">#&gt;    sample_name        known_barcode read_count read_count_proportion</span></span>
<span><span class="co">#&gt; 1:          8k AGGGGAGTCGCGTGGTAGGC      24298           0.015342523</span></span>
<span><span class="co">#&gt; 2:          8k GATGCCTGGAAAATACTTAA       4484           0.002831339</span></span>
<span><span class="co">#&gt; 3:          8k GTTTTGGTAGGTTCACCGGA       3676           0.002321142</span></span>
<span><span class="co">#&gt; 4:          8k AGAGCCTCACTAGCCTTTTG       3698           0.002335034</span></span>
<span><span class="co">#&gt; 5:          8k ATGACGGGATGCGCGTTAGT       2924           0.001846306</span></span>
<span><span class="co">#&gt; 6:          8k GTAAGCTCTAGGTGAAGAGT       2958           0.001867774</span></span>
<span><span class="co">#&gt;    projected_10000_confidence_0_8 projected_20000_confidence_0_8</span></span>
<span><span class="co">#&gt; 1:                            119                            241</span></span>
<span><span class="co">#&gt; 2:                             23                             46</span></span>
<span><span class="co">#&gt; 3:                             19                             37</span></span>
<span><span class="co">#&gt; 4:                             18                             36</span></span>
<span><span class="co">#&gt; 5:                             17                             27</span></span>
<span><span class="co">#&gt; 6:                             17                             28</span></span>
<span><span class="co">#&gt;    cum_sum_projected_10000_confidence_0_8 barcode_rank</span></span>
<span><span class="co">#&gt; 1:                                    119            1</span></span>
<span><span class="co">#&gt; 2:                                    142            2</span></span>
<span><span class="co">#&gt; 3:                                    161            3</span></span>
<span><span class="co">#&gt; 4:                                    179            4</span></span>
<span><span class="co">#&gt; 5:                                    196            5</span></span>
<span><span class="co">#&gt; 6:                                    213            6</span></span></code></pre></div>
<p>Ah, a table may not be of much use. Let’s draw a line plot to show
them instead.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">top_barcodes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">barcode_rank</span>, y<span class="op">=</span><span class="va">cum_sum_projected_10000_confidence_0_8</span>, </span>
<span>                         group<span class="op">=</span><span class="va">sample_name</span>, colour <span class="op">=</span> <span class="va">sample_name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span>linewidth<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://scales.r-lib.org/reference/pretty_breaks.html" class="external-link">pretty_breaks</a></span><span class="op">(</span>n<span class="op">=</span><span class="fl">10</span><span class="op">)</span>, labels <span class="op">=</span> <span class="fu"><a href="https://scales.r-lib.org/reference/label_number.html" class="external-link">label_comma</a></span><span class="op">(</span>accuracy <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_continuous</a></span><span class="op">(</span>breaks <span class="op">=</span> <span class="fu"><a href="https://scales.r-lib.org/reference/pretty_breaks.html" class="external-link">pretty_breaks</a></span><span class="op">(</span>n<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, hjust<span class="op">=</span><span class="fl">1</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    y <span class="op">=</span> <span class="st">"Number of cells"</span>,</span>
<span>    x <span class="op">=</span> <span class="st">"Barcode Ranking"</span>,</span>
<span>    title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Projected no. of cells for top"</span>, <span class="va">top_threshold</span>, <span class="st">"most abundant clones"</span><span class="op">)</span>,</span>
<span>    colour <span class="op">=</span> <span class="st">"Library"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="analyse_dnaseq_data_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Givanna Putri.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
